db/db_impl/db_impl.cc

```diff
@@ -1359,6 +1360,11 @@ Status DBImpl::ApplyReplicationLogRecord(ReplicationLogRecord record,
             // Ignore the update, we already have it
             continue;
           }
+          info->has_new_manifest_writes = true;
+          if (!allow_new_manifest_writes) {
+            // We don't expect new manifest writes, break early
+            break;
+          }
```

Comment:
Why not just return non-ok status in this case? It's a failure condition, right?

Reply:
This case will be treated as poison and what we should do is to cleanup local log and reopen shard. 

It feels weird to use any existing Status to represent this case. Or do you mean adding a new status in rocksdb?

Reply:
It feels weird to use `Corruption` as well since we need to distinguish this from the other corruption cases.

Agree current approach is weird. We can clean this up once we implement bumping mus once for each `kManifestWrite`.